### ISSUE TITLE
Allowing `<hr>` inside of `<select>`

### ISSUE THREAD

**Issue by jensimmons** [web-platform-tests/interop#573]
### Description

Safari 17 added support for Allowing `<hr>` inside of `<select>`. This makes it possible for developers to create simply lines inside dropdown menus by using semantic HTML, instead of styling hacks. It'd be very useful to have interoperability of this.  

### Specification

https://html.spec.whatwg.org/multipage/grouping-content.html#the-hr-element

### Tests

[Parser tests in html5lib_webkit02](https://wpt.fyi/results/html/syntax/parsing?label=master&label=experimental&aligned&q=html5lib_webkit02%20and%20%28subtest%3A5463526d91a8677b27b6967866d6605f1bb03aac%20or%20subtest%3A4879f476053094cf5602d325724675378856a902%20or%20subtest%3A44c88b90236f01ebc8e0123363b527640a07070c%20or%20subtest%3Acfb304e8f2d3cbdecc362226e7775cab452d5489%20or%20subtest%3A3fc625e7cb9b6ea72a9e252ede84c6fdd9680d87%20or%20subtest%3Aecd089f9b5193fad306c5b475c4711547fe5e209%20or%20subtest%3Acee2230c74671c594a1140a68d16e3d3e5ae005a%20or%20subtest%3A22b9fe36797d70a3b71a6aadc6ad7cff23c3fc90%20or%20subtest%3Aa82c3bf49c381b5f58c5c8a4bbbe0cef2458e28a%20or%20subtest%3A61f8d527795dc8044a95a3e2437de81e16597ceb%29)

### Current Implementations

- [ ] Blink
- [ ] Gecko
- [X] WebKit

### Browser bug reports

https://bugzilla.mozilla.org/show_bug.cgi?id=1830909 - Firefox Tracking Bug
https://bugs.chromium.org/p/chromium/issues/detail?id=1441883 - Chromium Tracking Bug (shipping in 119)

### Developer discussions

Discussed with great excitement on Shop Talk Show #585. 

### Workarounds

Workarounds include adding a "choice" to the list of items to select that's "--------------". Or attempting to apply CSS to the inside of a select menu.

### Accessibility Impact

This will allow developers to use proper semantic HTML to delineate sections in a drop-down menu, instead of creating fake selections that could be confusing.

### Privacy Impact

No privacy impact.

**Comment by lukewarlow** [comment]
Just as an FYI:

https://bugzilla.mozilla.org/show_bug.cgi?id=1830909 - Firefox Tracking Bug

https://bugs.chromium.org/p/chromium/issues/detail?id=1441883 - Chromium Tracking Bug (shipping in 119)

**Comment by mfreed7** [comment]
Tests are located at (a subset of):

https://wpt.fyi/results/html/syntax/parsing/html5lib_webkit02.html%3Frun_type=uri?label=master&label=experimental&aligned&q=5463526d91a8677b27b69

**Comment by gsnedders** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1830909 is now in autoland, so it seems likely this will soon be at 100% everywhere.

**Comment by jensimmons** [comment]
It probably doesn't make sense to include `<hr>` inside `<select>` in Interop 2024 — even if enthusiasm for it is high — if it's going to be 100% everywhere before Interop 2024 begins. In fact, if it had been 100% across the board in October, it might have been disqualified for that reason, since Interop Project focuses only on things that need work.

**Comment by gsnedders** [comment]
Thank you for proposing Allowing `<hr>` inside of `<select>` for inclusion in Interop 2024.

We wanted to let you know that this proposal was not selected to be part of Interop this year.

This proposal was not included in Interop 2024 because when we looked at its Web Platform Tests results, we found that it is already largely interoperable between tested browsers and does not need special coordinated attention.

For an overview of our process, see [proposal selection](https://github.com/web-platform-tests/interop/blob/main/2024/selection-process.md). Thank you again for contributing to Interop 2024!

_Posted on behalf of the Interop team._
